mautic/mautic
# Relative dates new

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ 4.x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9636 



#### Description:

Added new relative dates for segment filter,
- `'first day of '`
- `'last day of '`


#### Steps to test this PR:


1. Open this PR on Gitpod or pull-down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a segment with a filter by the relative date "2 days ago" or "3 months ago", etc.
3. Save and then go to edit the segment. Instead of a relative date, there will be an exact date and even a time
4. At first, the segment will work, but as soon as more than a relative date passes in time, contacts with an outdated date will not be deleted from the segment.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10821"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
